### PR TITLE
Feat: add drive macro IntoEvent

### PR DESCRIPTION
**Pull request by loloicci**
Based 0.16.2

### CODE REVIEW

**Comment by brew0722** [comment]
Implementation direction Overall, LGTM.

I have one question.
- In the case of `entry_point`, it is exposed as `cosmwasm-std` after adding it to the existing `cosmwasm-derive` package.
How about adding it in a similar way?
- Is there reason to add it to newer `cosmwasm-event-derive` package instead of adding it to `cosmwasm-derive` package?

File: packages/event-derive/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[package]
+name = "cosmwasm-event-derive"
+version = "0.16.2"
+edition = "2021"
```
Comment (brew0722) [inline packages/event-derive/Cargo.toml]: It seems like have to keep 2018 edition to get into cosmwasm standard items.

**Comment by loloicci** [comment]
> In the case of entry_point, it is exposed as cosmwasm-std after adding it to the existing cosmwasm-derive package.
How about adding it in a similar way?

In similar way, exported as a member of cosmwasm-std here.
https://github.com/loloicci/line-cosmwasm/blob/8b9f43fa34b5728e63f4c4a06e79d096254acd1c/packages/std/src/lib.rs#L118

> Is there reason to add it to newer cosmwasm-event-derive package instead of adding it to cosmwasm-derive package?

Few reasons, like to keep lib.rs simple and easiness of merging original side changes. No problem to move it to cosmwasm-derive package.

**Comment by brew0722** [comment]
Yes. Then it would be better to move to cosmwasm-derive.
As you know, it has the same meaning, and the more cargo pacakge, the more troublesome management seems to be.

**Comment by brew0722** [comment]
Oh, and I think we need  prepare a contract for the example in addition to the test code.
However, it is too cumbersome to develop individual contracts for every feature we develop. It would be good to discuss this separately.

**Comment by shiki-tak** [comment]
LGTM
By the way, what does "Into" mean?

**Comment by loloicci** [comment]
`IntoEvent` means the trait `Into<Event>`. The macro derives it from a struct.

**Pull request closed without merging**